selfservice/strategy/oidc/strategy_registration.go

```diff
@@ -200,26 +225,15 @@ func (s *Strategy) processRegistration(w http.ResponseWriter, r *http.Request, a
 			WithField("subject", claims.Subject).
 			Debug("Received successful OpenID Connect callback but user is already registered. Re-initializing login flow now.")
 
-		// If return_to was set before, we need to preserve it.
-		var opts []login.FlowOption
-		if len(a.ReturnTo) > 0 {
-			opts = append(opts, login.WithFlowReturnTo(a.ReturnTo))
-		}
-
-		// This endpoint only handles browser flow at the moment.
-		ar, _, err := s.d.LoginHandler().NewLoginFlow(w, r, flow.TypeBrowser, opts...)
+		lf, err := s.registrationToLogin(w, r, a, provider.Config().ID)
 		if err != nil {
-			return nil, s.handleError(w, r, a, provider.Config().ID, nil, err)
+			return nil, err
```

Comment:
This could lead to confusion down the line. Wouldn't it be clearer, if the error is handled here (call `s.handleError`) and just return the error from `registrationToLogin`? Do we have some kind of "guidelines" on this?

I noticed this pattern a lot and have always wondered why we handle these errors on the deepest level, instead of "at the top of the strategy" (e.g. entry point of the strategy - here the `Register` method).

Reply:
Not sure I follow, do you want me to call `s.handleError` here instead of just returning the error from the `registrationToLogin` function? 

In such as case, I don't think it is possible, since the function adds data to the registration flow (such as error messages) and then passes that along to `handleError` which determines redirect strategies.